### PR TITLE
sturdier check

### PR DESCRIPTION
**Pull request by witmicko**
enhance [#678](https://github.com/MetaMask/mobile-planning/issues/678)

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by NEllusion — COMMENTED** [review]
A couple nitpicky comments, but otherwise the solution looks good. Running the branch now to experiment with locally. Can we add a unit test as well that this works as expected for a couple payloads?

**Review by NEllusion — APPROVED** [review]

**Review by NEllusion — APPROVED** [review]
Tested locally and changes look good to me

File: app/util/linkCheck.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { PhishingController as PhishingControllerClass } from '@metamask/phishing-controller';
+import Engine from '../core/Engine';
+const ALLOWED_PROTOCOLS = ['http:', 'https:'];
+const DENYLISTED_DOMAINS = ['metamask.app.link'];
+
```
Comment (Cal-L) [inline app/util/linkCheck.ts:5]: Can we use URL from
`import URL from 'url-parse';`. We've had issues with malformed URLs in the past and url-parse catches the issues in those scenarios.

Comment (Cal-L) [reply app/util/linkCheck.ts:5]: Once this is added, could we add a test case for a malformed url?

File: app/util/linkCheck.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    // expect(
+    //   isLinkSafe(
+    //     'https://metamask.app.link/send/pay-Contract-Address@chain-id/transfer?address=Receiver-Address&uint256=1e21',
+    //   ),
+    // ).toEqual(false);
+
+    // expect(isLinkSafe('javascript:alert(1)')).toEqual(false);
+    // expect(isLinkSafe('j&Tab;avascript:alert(1);')).toEqual(false);
+    // expect(isLinkSafe('&Tab;javascript:alert(1);&tab;')).toEqual(false);
+    // expect(isLinkSafe('javas\x00cript:javascript:alert(1)')).toEqual(false);
```
Comment (Cal-L) [inline app/util/linkCheck.test.ts:30]: Were these tests removed on purpose?

File: app/util/linkCheck.ts (hunk 1 above)
Comment (Cal-L) [reply app/util/linkCheck.ts:7]: @witmicko ^ 
Could we add a test along the lines of 
`expect(isLinkSafe('htps://ww.example.com/')).toEqual(true);`

**Review by Cal-L — APPROVED** [review]
Could you pull from main to resolve the checks - Aside from that, LGTM
Please check in with QA @cortisiko or @chrisleewilcox for QA hand off before merging

**Comment by Cal-L** [comment]
@SocketSecurity ignore @types/is-url@1.2.30

**Comment by Cal-L** [comment]
@SocketSecurity ignore @types/url-parse@1.4.8

**Review by cortisiko — APPROVED** [review]
Looks good!

**Pull request merged**